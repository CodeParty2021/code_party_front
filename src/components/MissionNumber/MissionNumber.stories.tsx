import React from "react";
import { ComponentStory, ComponentMeta } from "@storybook/react";

import { MissionNumber } from "./MissionNumber";

export default {
  title: "components/MissionNumber",
  component: MissionNumber,
} as ComponentMeta<typeof MissionNumber>;

const Template: ComponentStory<typeof MissionNumber> = (args) => (
  <MissionNumber {...args} />
);

export const One = Template.bind({});
One.args = {
  number: 1,
};

export const Two = Template.bind({});
Two.args = {
  number: 2,
};

export const Three = Template.bind({});
Three.args = {
  number: 3,
};
