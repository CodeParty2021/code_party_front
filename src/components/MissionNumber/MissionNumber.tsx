import React from "react";
import {
  MissionNumberBGStyle,
  MissionNumberStyle,
  MissionNumberStyleProps,
} from "./MissionNumberStyle";

type Props = MissionNumberStyleProps & {
  number: number;
};

export const MissionNumber: React.FC<Props> = ({ number, ...styleProps }) => {
  return (
    <MissionNumberStyle {...styleProps}>
      <MissionNumberBGStyle src="/img/mission_number_bg.svg" wrapper="svg" />
      <span>{number}</span>
    </MissionNumberStyle>
  );
};
