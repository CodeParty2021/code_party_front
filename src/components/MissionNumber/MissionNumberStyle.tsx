import { ReactSVG } from "react-svg";
import styled, { css } from "styled-components";
import { missionNumber } from "styles/colors";
import { FONT } from "styles/constants/constants";

export type MissionNumberStyleProps = {};

const defaultStyle = css`
  font-family: ${FONT.F_851Gkktt};
  font-size: 30px;
  color: ${missionNumber.font};
  overflow: hidden;

  width: 50px;
  height: 59px;
  position: relative;

  & > span {
    position: absolute;
    left: 50%;
    top: 12px;
    transform: translateX(-50%);
  }
`;

export const MissionNumberStyle = styled.div<MissionNumberStyleProps>`
  ${defaultStyle}
`;

MissionNumberStyle.defaultProps = {};

export const MissionNumberBGStyle = styled(ReactSVG)`
  position: absolute;
  left: 0;
  top: 0;
  width: 100%;
  height: 100%;
  fill: ${missionNumber.bg};
`;
