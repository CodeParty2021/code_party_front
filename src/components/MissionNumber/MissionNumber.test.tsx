import React from "react";
import { MissionNumber } from "./MissionNumber";
import "jest-styled-components";
import { render } from "enzyme";

describe("<MissionNumber />", () => {
  it("snapshot test", () => {
    const wrapper = render(<MissionNumber number={1} />);

    expect(wrapper).toMatchSnapshot();
  });
});
