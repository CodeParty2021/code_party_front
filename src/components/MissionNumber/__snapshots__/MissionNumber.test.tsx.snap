// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MissionNumber /> snapshot test 1`] = `
.c0 {
  font-family: "851Gkktt";
  font-size: 30px;
  color: #F5F6F8;
  overflow: hidden;
  width: 50px;
  height: 59px;
  position: relative;
}

.c0 > span {
  position: absolute;
  left: 50%;
  top: 12px;
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
}

.c1 {
  position: absolute;
  left: 0;
  top: 0;
  width: 100%;
  height: 100%;
  fill: #2A2C33;
}

<div
  class="c0"
>
  <svg
    class="c1"
    xlink="http://www.w3.org/1999/xlink"
    xmlns="http://www.w3.org/2000/svg"
  />
  <span>
    1
  </span>
</div>
`;
